app/models/conference.rb

```diff
@@ -595,6 +595,20 @@ def registration_limit_exceeded?
     registration_limit > 0 && registrations.count >= registration_limit
   end
 
+  # Returns an hexadecimal color given a collection. The returned color changed
+  # when the number of element in the collection changes and for consecutive
+  # number of elements it returns highly different colors.
+  def next_color_for_collection(collection)
+    # we have different start indices for every collection to generate a
+    # different color for every of them.
+    collections = {
+      tracks: { collection: program.tracks, start_index: 1 },
+      levels: { collection: program.difficulty_levels, start_index: 51 },
+      types: { collection: program.event_types, start_index: 101 }
+    }
+    next_color(collections[collection][:collection].count + collections[collection][:start_index])
```

Comment:
Why not calculate 

``
collection.count + start_index
`` 

in the hash? Doing it that way you don't need a nested hash. Plus you could name the hash ``start_index`` and just call 

```
next_color(start_index[:collection])
```

which makes it more easy to understand?